openedx_cas/backends.py

```diff
@@ -1,41 +1,133 @@
-"""Centralized Authentication Service backend based on the social-core library
-implementation.
+"""Centralized Authentication Service backend based on the social-core library implementation.
+
+Implements two backends, CASBackendOverride that allows us to perform the changes to the cas_validation
+method, making it compatible with the BaseAuth class.
 """
 from social_core.backends.base import BaseAuth
+from social_core.exceptions import AuthFailed
+from django.conf import settings
+from django_cas_ng.backends import CASBackend
+import urllib.parse
+from django_cas_ng.utils import get_cas_client
+from django.contrib.auth import get_user_model
+
+import logging
+
+UserModel = get_user_model()
 
+logger = logging.getLogger(__name__)
 
-class CASAuth(BaseAuth):
+class CASAuth(BaseAuth, CASBackend):
     """CAS authentication backend base class.
 
     Settings will be inspected to get more values names that should be
     stored on extra_data field. The setting name is created following the
     pattern SOCIAL_AUTH_<uppercase current backend name>_EXTRA_DATA.
 
-    access_token is always stored.
-
     URLs settings:
-        AUTHORIZATION_URL       Authorization service url
-        ACCESS_TOKEN_URL        Access token URL
+        CAS_SERVICE_URL:
+            Our authorization service url.
+        CAS_CLIENT_VERSION:
+            Version of client protocol.
+            Currently V2 and V3 versions are supported, not sure how V1 will perform
+        CAS_SERVER_URL:
+            Server URL against the user is authenticated
     """
+
     name = "centralized-auth-service"
 
     def auth_url(self):
         """Get the URL to which we must redirect in order to
         authenticate the user:
 
-        Redirect /CAS-provider-domain/
+        Redirect /CAS-service-domain/
         """
-        raise NotImplementedError("Not implemented yet")
+
+        return f"{settings.CAS_SERVER_URL}?service={urllib.parse.quote(settings.CAS_SERVICE_URL)}"
 
     def auth_complete(self, *args, **kwargs):
         """Completes login process, must return user instance.
 
-        We could use the implementation of:
-        https://github.com/django-cas-ng/django-cas-ng/blob/master/django_cas_ng/backends.py#L22
-        And refactor it as we see fit according our use case.
-
         Request:
-        GET
-	    https://CAS-provider-domain/auth/complete/?ticket=ticket
+        GET https://CAS-service-domain/auth/complete/?ticket=ticket
+        :returns: [User] Authenticated User object or None if authenticate failed
+        """
+
+        request = kwargs["request"]
+        ticket = request.GET["ticket"]
+        response = self.cas_validation(request, ticket, settings.CAS_SERVICE_URL)
+        kwargs.update({"response": response, "backend": self})
+
+        logger.info(f"User {response} logged in with {settings.CAS_SERVER_URL}")
```

Comment:
@mariajgrimaldi I'm not sure if this log is useful (was for development purposes). Does the third_party_app or pipeline logs this information?